### DISCUSSION TITLE
Bypass asking Initramfs password

### DISCUSSION THREAD

**Discussion by loxK** [openzfs/zfs#13655] [Q&A]
Hello,

It is related to the [initramfs contrib](https://github.com/openzfs/zfs/tree/master/contrib/initramfs).
On an Ubuntu encrypted root on ZFS installation, the ZFS key is stored in a LUKS encrypted zvol (`rpool/keystore`).

I need to hook a bash script that take care of opening the luks zvol so that password is not asked. If my script fails, password should be asked. Is that possible ?

**Comment by mskarbek** [comment]
You can take a look at [Clevis](https://github.com/latchset/clevis).
